packages/expo-cli/src/commands/run/android/runAndroid.ts

```diff
@@ -101,9 +102,16 @@ async function resolveOptionsAsync(
     port = 8081;
   }
 
-  const variant = options.variant.toLowerCase();
-  const apkDirectory = Android.getAPKDirectory(projectRoot);
-  const apkVariantDirectory = path.join(apkDirectory, variant);
+  // TODO: why would this be different? Can we get the different name?
+  const appName = 'app';
+
+  const apkDirectory = path.join(projectRoot, 'android', appName, 'build', 'outputs', 'apk');
+
+  // buildDeveloperTrust -> build, developer, trust (where developer, and trust are flavors).
+  const [buildType, ...flavorDimensions] = options.variant
```

Comment:
Naming here is partially incorrect, if you have command `:app:buildFooBarRelease`
- app is an android module
- `foo` and `bar` are flavors (not flavorDimensions), but actual flavours can be named `Foo` and `Bar` we don't know that just based on the gradle command
-  flavorDimensions defines order of the flavors in the command, so if you have in your build.gradle `flavourDimensions: "dim1" "dim2"` then `foo` has to have dimension set to `dim1` and `bar` to `dim2`
- variant in this example would be `FooBarRelease`, build/bundle/assemble... is not the part of the variant
- buildType is either debug, release or some custom value

Reply:
renamed `flavorDimensions` to `flavors`